string_encode.py

```diff
@@ -305,18 +305,15 @@ def encode(self, text):
         return hasher.hexdigest()
 
 
-class Escaper(StringEncode):
+class EscapeRegexCommand(StringEncode):
+    regex = re.compile(r'(?<!\\)([?\\*.+^$()\[\]\{\}\|])')
 
     def encode(self, text):
-        return re.sub(r'(?<!\\)(%s)' % self.meta, r'\\\1', text)
+        return self.regex.sub(r'\\\1', text)
 
 
-class EscapeRegexCommand(Escaper):
-    meta = r'[?\\*.+^$()\[\]\{\}\|]'
-
-
-class EscapeLikeCommand(Escaper):
-    meta = r'[%_]'
+class EscapeLikeCommand(EscapeRegexCommand):
```

Comment:
I'd prefer a common parent class, but I can make that change. I like the precompiled regex 👍

Reply:
Actually with only two commands an trivial implementation, I'd even just copy `run` method, TBH.

Reply:
Pushed a change accordingly. I'd inherit from a base class only when more complex implementations or many sub-commands existed.